### DISCUSSION TITLE
How to show relation data in table view ? Admin Panel v2.x

### DISCUSSION THREAD

**Discussion by wahyusa** [filamentphp/filament#1809] [Help] [answered]
Hi, thanks for this great package!

I've just try Admin Panel Filamentphp v2.x and it works perfectly but I don't know how to show relation data in a table like this ?

![Post hasOne category hasOne author](https://images2.imgbox.com/e4/04/b46SyK3T_o.png)

I've try to use [relationmanager ](https://filamentphp.com/docs/2.x/admin/resources#relations) but it just add new section to manage category in "Posts Edit" page, which is not what I need.. because I can use select with search to change category_id from a posts.

On table view it show nothing except blank content

Expect :
Posts hasOne category_name and hasOne author_name in table view
instead of category_id or author_id

**Comment by wahyusa** [comment, marked as answer]
Sorry finally I've found it 

[https://filamentphp.com/docs/2.x/tables/columns#setting-a-default-value](https://filamentphp.com/docs/2.x/tables/columns#setting-a-default-value)

Columns may be created using the static make() method, passing its name. The name of the column should correspond to a column or accessor on your model. You may use "dot syntax" to access columns within relationships.

```php

use Filament\Tables\Columns\TextColumn;
 
TextColumn::make('title')
 
TextColumn::make('author.name')
```

**Comment by indigoram89** [comment]
How to do it with a link to that relation? Like Laravel Nova does it automaticaly.

**Reply by danharrin** [reply]
`->url(fn ($record): string => RelatedResourceName::getUrl('edit', ['record' => $record->relationshipname]))`

We don't do many things automatically, this is a UI framework and not a "generator" like Nova

**Reply by indigoram89** [reply]
I understood you and we (our developers and I) very thank to you.

Also we are sure that Filament is much better that Nova.

One missed thing is it will be convenient to have a TableColumn like:

```
RelationColumn::make('Name')
    ->relation('relationName')
     ->resource(RelatedResourceName::class)
```

...which do your example under hood.

**Reply by danharrin** [reply]
It would have to be outside the table builder, as the table builder does not depend on the panel builder at all. Maybe you should create a plugin, or just a class in your app to do it :)

**Reply by indigoram89** [reply]
Sure, we will do it. Thank you! :)
